crypto/hashes.go

```diff
@@ -0,0 +1,150 @@
+// Copyright (C) 2019-2022 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package crypto
+
+import (
+	"crypto/sha512"
+	"errors"
+	"fmt"
+	"hash"
+
+	"github.com/algorand/go-algorand/protocol"
+	"github.com/algorand/go-sumhash"
+)
+
+// HashType represents different hash functions
+type HashType uint16
+
+// IsValid verifies that the hash type is in a valid range.
+func (h HashType) IsValid() error {
```

Comment:
this isn't really a blocker, but having a method called "IsValid()" that returns an `error` doesn't seems right.
I would expect it to return a bool. If the method would have been called "Validate()", then it would be just fine.